sdk/src/transaction.rs

```diff
@@ -330,6 +334,27 @@ impl Transaction {
         }
     }
 
+    pub fn verify_secp256k1(&self) -> Result<()> {
+        for instruction in &self.message().instructions {
+            if instruction.program_id_index as usize >= self.message().account_keys.len() {
+                return Err(TransactionError::AccountNotFound);
+            }
+            let program_id = &self.message().account_keys[instruction.program_id_index as usize];
+            if crate::secp256k1_program::check_id(program_id) {
```

Comment:
We will probably have more of these, could generalize the verify structure to accommodate additions in the future

Reply:
I didn't find a way to do like a match Pubkey, so I think I'll punt on this for now, although it does seem worthwhile to have some kind of framework here. I might be the one adding the next one anyway...